lib/mongo/lib/compensateForLatency.js

```diff
@@ -1,27 +0,0 @@
-import RedisSubscriptionManager from '../../redis/RedisSubscriptionManager';
-import { Events, RedisPipe } from '../../constants';
-/**
- * Latency compensator acts exactly as a synthetic event, which is very quick
- *
- * @param channels
- * @param collectionName
- * @param event
- * @param doc
- * @param fields
- */
-export default (channels, collectionName, event, doc, fields) => {
-    channels.forEach(channel => {
-        RedisSubscriptionManager.process(channel, {
-            [RedisPipe.DOC]: doc,
-            [RedisPipe.EVENT]: event,
-            [RedisPipe.FIELDS]: fields
-        });
-
-        const dedicatedChannel = collectionName + '::' + doc._id;
```

Comment:
I noticed that the `dedicatedChannel` block is nested within `channels.forEach`..

This means if `channels.length == n` then it will publish to `dedicatedChanel` `n` times.

Reply:
Good catch, that was bad, luckily, rarely we had multiple channels to push to, most of the times it was just one.